### PR TITLE
Run apiview checks in pr to detect public api changes

### PR DESCRIPTION
**Pull request by m-nash**
Fixes https://github.com/microsoft/typespec/issues/3203
Fixes https://github.com/microsoft/typespec/issues/3207

### CODE REVIEW

**Review by hallipr — APPROVED** [review]

File: packages/http-client-csharp/package.json [hunk 1]
```diff
@@ -34,12 +34,16 @@
...
+    "format": "pnpm -w format:dir packages/http-client-csharp",
+    "extract-api": "npx api-extractor run --local --verbose"
   },
   "files": [
     "dist/**"
   ],
   "dependencies": {
+    "@azure-tools/ts-genapi": "^1.0.7",
+    "@microsoft/api-extractor-model": "^7.28.2",
+    "js-tokens": "^8.0.0",
```
Comment (timotheeguerin) [inline packages/http-client-csharp/package.json:46]: are those not supposed to be dev dependencies

File: packages/http-client-csharp/package.json [hunk 2]
```diff
@@ -34,12 +34,16 @@
...
     "lint:fix": "eslint . --fix",
-    "format": "pnpm -w format:dir packages/http-client-csharp"
+    "format": "pnpm -w format:dir packages/http-client-csharp",
+    "extract-api": "npx api-extractor run --local --verbose"
   },
   "files": [
     "dist/**"
   ],
   "dependencies": {
+    "@azure-tools/ts-genapi": "^1.0.7",
```
Comment (timotheeguerin) [inline packages/http-client-csharp/package.json:44]: where is this package coming from its not on npm

File: packages/http-client-csharp/package.json (hunk 1 above)
Comment (timotheeguerin) [reply packages/http-client-csharp/package.json:46]: are they even used at all? I see `"@microsoft/api-extractor": "^7.40.3",` as needed for the `api-extractor` command but not sure what those do

File: packages/http-client-csharp/package.json (hunk 2 above)
Comment (m-nash) [reply packages/http-client-csharp/package.json:44]: it comes from running the extract command locally

Comment (m-nash) [reply packages/http-client-csharp/package.json]: I think the package itself is from our internal feed

Comment (m-nash) [reply packages/http-client-csharp/package.json]: Ok, confirmed those only get added when you run Generate-APIView-CodeFile.ps1.  This script needs them to convert the *.api.json file into a file apiview can consume *_js.json.

Comment (timotheeguerin) [reply packages/http-client-csharp/package.json]: yeah it be great if that work with them not included in your last change as this basically now makes it impossible to contribute for an external user.

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**